src/features/unified-share-modal/SharedLinkPermissionMenu.js

```diff
@@ -33,11 +33,19 @@ type Props = {
     },
 };
 
-class SharedLinkPermissionMenu extends Component<Props> {
+type State = {
+    hasSeenEditTag: boolean,
+};
+
+class SharedLinkPermissionMenu extends Component<Props, State> {
     static defaultProps = {
         trackingProps: {},
     };
 
+    state = {
+        hasSeenEditTag: false,
```

Comment:
I was under the impression that the FTUX targeting API stored this state internally and exposed helpers to help mange it. It may be worth reaching out to Product Adoption for some guidance. Maybe @megansmith-box could help?

Reply:
You are correct about the FTUX targeting API internal state. I talked with @shahzadaziz and we decided that the desired behavior was to show the "New" tag when the tooltip is shown. The main principle of the targeting API is that it only shows one onboarding behavior at a time. However, in this case we want to show two so this was the workaround for that. Definitely interested if there is a better way.

Reply:
I agree that the main issue here is that the way the generic experience machine is set up, only one experience can be shown at a time per page load. Therefore, either the Tooltip or the Tag will show up independently of each other. I do see this implementation as a work around. 

I believe a better way would be to remove `shared-link-with-edit-tooltip` & `shared-link-with-edit-tag` from the generic experience machine and create a custom 2 step state machine for this experience. That way they are grouped together since they are dependent of each other. The flow for the new machine would be
idle -> show tooltip -> show tag -> done

There is some documentation about this here: https://confluence.inside-box.net/display/ETO/4+-+Creating+Tooltip+Flows and several examples of custom state machines in EUA. 

Let me know what you think!